Views/VMDisplayMetalViewController+Pointer.m

```diff
@@ -0,0 +1,116 @@
+    [self addInteractionTo:self.pauseResumeButton];
+    [self addInteractionTo:self.restartButton];
+    [self addInteractionTo:self.zoomButton];
+    [self addInteractionTo:self.toggleKeyboardButton];
+    [self addInteractionTo:self.configureButton];
+    [self addInteractionTo:self.hideToolbarButton];
+    [self addInteractionTo:self.resumeBigButton];
+}
+
+#pragma mark - UIPointerInteractionDelegate
+- (UIPointerStyle *)pointerInteraction:(UIPointerInteraction *)interaction styleForRegion:(UIPointerRegion *)region {
+    // Hide cursor while hovering in VM view
+    if (interaction.view == self.mtkView) {
+        return [UIPointerStyle hiddenPointerStyle];
+    } else {
+        return [UIPointerStyle styleWithShape:[UIPointerShape shapeWithRoundedRect:interaction.view.bounds] constrainedAxes:UIAxisBoth];
+    }
+}
+
+static CGFloat CGPointToPixel(CGFloat point) {
+    return point * [UIScreen mainScreen].scale; // FIXME: multiple screens?
+}
+
+- (bool)isPointOnVMDisplay:(CGPoint)pos {
+    CGSize screenSize = self.mtkView.drawableSize;
+    CGSize scaledSize = {
+        self.vmDisplay.displaySize.width * self.vmDisplay.viewportScale,
+        self.vmDisplay.displaySize.height * self.vmDisplay.viewportScale
+    };
+    CGRect drawRect = CGRectMake(
+        self.vmDisplay.viewportOrigin.x + screenSize.width/2 - scaledSize.width/2,
+        self.vmDisplay.viewportOrigin.y + screenSize.height/2 - scaledSize.height/2,
+        scaledSize.width,
+        scaledSize.height
+    );
+    pos.x -= drawRect.origin.x;
+    pos.y -= drawRect.origin.y;
+    if (pos.x < 0) {
+        return false;
+    } else if (pos.x > scaledSize.width) {
+        return false;
+    }
+    if (pos.y < 0) {
+        return false;
+    } else if (pos.y > scaledSize.height) {
+        return false;
+    }
+    return true;
+}
+
+
+- (UIPointerRegion *)pointerInteraction:(UIPointerInteraction *)interaction regionForRequest:(UIPointerRegionRequest *)request defaultRegion:(UIPointerRegion *)defaultRegion {
+    // Requesting region for the VM display?
+    if (interaction.view == self.mtkView) {
+        // Then we need to find out if the pointer is in the actual display area or outside
+        CGPoint location = [request location];
+        CGPoint translated = location;
+        translated.x = CGPointToPixel(translated.x);
+        translated.y = CGPointToPixel(translated.y);
+        
+        if ([self isPointOnVMDisplay:translated]) {
+            // move vm cursor, hide iOS cursor
+            CGPoint clipped = [self clipCursorToDisplay:translated];
+            [_cursor setCenter:clipped];
+            return [UIPointerRegion regionWithRect:[self.mtkView bounds] identifier:@"vm view"];
+        } else {
+            // don't move vm cursor, show iOS cursor
+            return nil;
+        }
+    } else {
+        // Is it a UIControl (e. g. Button)?
+        if ([interaction.view.class respondsToSelector:NSSelectorFromString(@"isEnabled")]) {
```

Comment:
Are you sure this is the right check? From a cursory glance it seems like you're using the wrong receiver…also, can you not just use `@selector`?

Reply:
It's a hack, and has been removed!